### PR TITLE
adjustments to go mod and logger + ES + minor refactors + new redis function

### PR DESCRIPTION
**Pull request by ArpinFidel**
refactor: use one go mod
refactor: clean up logger code
refactor: add trace to logger
refactor: LogErrorConsumer -> Log
refactor: allow empty struct on redis key (give warning)
refactor: redis key on pagination
remove other pagination
fix: add check for uninitialized headers on request context

### CODE REVIEW

**Comment by Ahtuz** [comment]
restid dienyahkan gak nih? 🤣

File: logger/logger.go [hunk 1]
```diff
@@ -49,84 +45,82 @@ func realIP(req *http.Request) string {
...
 
-// LogError for api
-func LogError(c *gin.Context, uuid, errMsg string) {
-	w := c.Writer
-	r := c.Request
-	log := logrus.New()
-	client, err := elastic.NewClient(elastic.SetSniff(false), elastic.SetURL(os.Getenv("ELASTICSEARCH_HOST")))
+	client, err := elastic.NewClient(
+		elastic.SetSniff(false),
+		elastic.SetURL(os.Getenv("ELASTICSEARCH_HOST")))
```
Comment (Amnestia) [inline logger/logger.go:53]:
```suggestion
		elastic.SetURL(os.Getenv("ELASTICSEARCH_HOST")),
		)
```

File: logger/logger.go [hunk 2]
```diff
@@ -49,84 +45,82 @@ func realIP(req *http.Request) string {
...
+		logger.Println(errMsg)
 		return
 	}
 
-	hook, err := elogrus.NewAsyncElasticHook(client, os.Getenv("ELASTICSEARCH_HOST"), logrus.DebugLevel, "service-logger")
+	hook, err := elogrus.NewAsyncElasticHook(
+		client,
+		os.Getenv("ELASTICSEARCH_HOST"),
+		logrus.DebugLevel,
+		service)
```
Comment (Amnestia) [inline logger/logger.go:64]:
```suggestion
		service,
		)
```

**Review by Amnestia — COMMENTED** [review]
kalau misalkan bentuknya dibikin kebawah, mungkin lebih baik kalau kurungnya di kebawahin juga

**Review by heggies — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,22 @@
+module github.com/forkyid/go-utils
+
+go 1.14
+
+require (
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible
+	github.com/gin-gonic/gin v1.6.3
+	github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8
+	github.com/go-playground/validator v9.31.0+incompatible
```
Comment (Ahtuz) [inline go.mod:9]: validator perlu di update ke v10 gak?
https://godoc.org/gopkg.in/go-playground/validator.v10

**Review by Ahtuz — APPROVED** [review]

**Review by Ahtuz — APPROVED** [review]

**Review by yafetzz — APPROVED** [review]

**Pull request merged**